bin/jira.py

```diff
@@ -0,0 +1,331 @@
+            p = Pulse(**yaml.safe_load(f))
+
+        log("Creating new pulse in Jira...")
+        pulse_id = self._new_pulse(p)
+        self._create_and_add_issues(p, pulse_id)
+
+    def add_to_pulse(self, path: str):
+        with open(os.path.expanduser(args.path)) as f:
+            p = Pulse(**yaml.safe_load(f))
+
+        log(f"Fetching ID of existing pulse: '{p.pulse_name}'...")
+        pulse_id = self._get_pulse_id(p.board_id, p.pulse_name)
+        log("Pulse ID found")
+        self._create_and_add_issues(p, pulse_id)
+
+    def _create_and_add_issues(self, p: Pulse, pulse_id: int):
+        log(f"Creating {len(p.issues)} issues...")
+        issues = []
+        for issue in p.issues:
+            key = self._new_issue(
+                issue, p.backlog, p.shared_labels, p.shared_components
+            )
+            issues.append(key)
+
+        log("Adding issues to pulse")
+        issues.extend(p.existing_issues)
+        self._add_issues_to_pulse(issues, pulse_id)
+        log("Done")
+
+    def print_epics(self, project: str):
+        jql = f"project={project} AND issuetype=Epic AND statusCategory!=Done"
+        start = 0
+        page_size = 100
+
+        log(f"Pulling Epic details for {project}...")
+        while True:
+            payload = {
+                "jql": jql,
+                "fields": ["summary"],
+                "maxResults": page_size,
+                "startAt": start
+            }
+            resp = requests.post(
+                f"{JIRA_URL}/rest/api/2/search",
+                json=payload,
+                auth=self._auth()
+            )
+            resp.raise_for_status()
+            data = resp.json()
+
+            if len(data["issues"]) == 0:
+                return
+
+            for issue in data["issues"]:
+                print(f"{issue["key"]}\t{issue["fields"]["summary"]}")
+            start += page_size
+
+    def _auth(self):
+        return (self._credentials.user, self._credentials.token)
+
+    def _new_pulse(self, p: Pulse) -> int:
+        end_date = p.start_date + timedelta(days=p.duration_days)
+        payload = {
+            "name": p.pulse_name,
+            "goal": p.pulse_goal,
+            "originBoardId": p.board_id,
+            "startDate": p.start_date.isoformat(),
+            "endDate": end_date.isoformat(),
+        }
+        resp = requests.post(
+            f"{JIRA_URL}/rest/agile/1.0/sprint",
+            json=payload,
+            auth=self._auth()
+        )
+        resp.raise_for_status()
+        id = resp.json()["id"]
+
+        return id
+
+    def _new_issue(
+        self,
+        i: Issue,
+        backlog: str,
+        labels: list[str],
+        components: list[str],
+        fix_versions: list[str]
+    ) -> str:
+        i.labels.extend(labels)
+        i.components.extend(components)
+        i.fix_versions.extend(fix_versions)
+
+        payload = {
+            "fields": {
+                "asignee": None,
```

Comment:
and `'asignee' cannot be set. It is not on the appropriate screen, or unknown.` , there is a typo in the keyword